### ISSUE TITLE
Duplicate regex Matcher

### ISSUE THREAD

**Issue by alb-i986** [hamcrest/JavaHamcrest#293]
Hi,

First of all, thanks for bringing this project back to life.

If I'm not mistaken there's a duplication in the regex matchers:
- MatchesPattern
- StringRegularExpression

There were none for such a long time and now there's two LOL
I'm not complaining huh, but I think it would be wise to leave only one ;)

Cheers

**Comment by tingtron** [comment]
More specifically, there are two ways to match regex in test code:
1.  org.hamcrest.Matchers.matchesRegex(regexString)
2. org.hamcrest.text.MatchesPattern.matchesPattern(regexString)

They are semantically the same, and mostly similar in implementation.

**Comment by nhojpatrick** [comment]
Yes the classes do appear to be duplicates and cover very similar functionaility;

`org.hamcrest.Matchers.matchesRegex(String)` basically calls out to `org.hamcrest.core.StringRegularExpression`

`org.hamcrest.core.StringRegularExpression`
`org.hamcrest.text.MatchesPattern`

**Comment by hakanai** [comment]
I also wonder about:
* `theInstance(T)` vs `sameInstance(T)`
* `any(Class<T>)` vs `instanceOf(Class<T>)`

**Comment by martinandersson** [comment]
Similarly, there's [`Matchers.matchesPattern(Pattern)`](https://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/Matchers.html#matchesPattern-java.util.regex.Pattern-) and [`Matchers.matchesRegex(Pattern)`](https://hamcrest.org/JavaHamcrest/javadoc/2.1/org/hamcrest/Matchers.html#matchesRegex-java.util.regex.Pattern-).

This is confusing and wasting time for people trying to learn. I recommend deleting `matchesRegex` 👍
